### PR TITLE
chore(postinst): Delete unnecessary postinst code

### PR DESCRIPTION
**Pull request by holmanb**
## Proposed Commit Message
```
chore(postinst): Delete unnecessary postinst code
```

## Additional Context

1) Delete version gated code
- most recent version gate was 22.1-14-g2e17a0d6-0ubuntu1~22.04.5
- If someone somehow upgraded a Jammy instance to noble without getting the newer Jammy cloud-init package available in jammy-updates, the fix in the postinst would still run since this code exists in all releases of noble.

2) Delete old preseed code. Debian doesn't use our postinst, and we don't use preseed, so just drop it -> we can probably do this with the maas preseed code too but that requires more investigation

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: debian/cloud-init.postinst [hunk 1]
```diff
@@ -109,22 +109,6 @@ for k in sys.argv[2:]:
    db_unregister "cloud-init/maas-metadata-credentials" || :
 }
 
-handle_preseed_local_cloud_config() {
```
Comment (TheRealFalcon) [inline debian/cloud-init.postinst:112]: The call site needs to be removed too

File: debian/cloud-init.postinst [hunk 2]
```diff
@@ -390,23 +216,6 @@ EOF
...
-
-   remove_deprecated_cloud_config_modules "$2"
-
-   fix_azure_upgrade_1611074 "$2"
-
-   cleanup_ureadahead "$2"
-   fix_lp1889555 "$2"
-   change_cloud_init_output_log_permissions "$2"
-
-   rename_hook_hotplug_udev_rule
```
Comment (TheRealFalcon) [inline debian/cloud-init.postinst:404]: Did you mean to include this one? The definition hasn't been deleted and this is a relatively recent addition.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
I left some inline comments.

If we're confident there's no way we can install with preseed anymore, than I'm not sure why we can't also delete the MAAS preseed code.

File: debian/cloud-init.postinst (hunk 2 above)
Comment (holmanb) [reply debian/cloud-init.postinst:404]: No I didn't - fixed.

**Comment by holmanb** [comment]
> If we're confident there's no way we can install with preseed anymore, than I'm not sure why we can't also delete the MAAS preseed code.

+1 agreed

@TheRealFalcon I think that I've addressed all of your comments.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Two more inline.

File: debian/cloud-init.postinst (hunk 2 above)
Comment (TheRealFalcon) [reply debian/cloud-init.postinst:404]: I'm not seeing it.

**Comment by TheRealFalcon** [comment]
@holmanb , just because it is showing up under the resolved comments, I wanted to make sure you can see my latest comment: https://github.com/canonical/cloud-init/pull/5487#discussion_r1686522844

**Pull request merged**